GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -424,35 +431,29 @@ private void Update()
                     {
                         try
                         {
-                            GitExtUtils.ArgumentString cmd = GitCommandHelpers.GetAllChangedFilesCmd(true, UntrackedFilesMode.Default, noLocks: noLocks);
-                            ExecutionResult result = await module.GitExecutable.ExecuteAsync(cmd, cancellationToken: cancelToken).ConfigureAwait(false);
-
-                            if (result.ExitedSuccessfully && !ModuleHasChanged())
-                            {
-                                // Update callers also if cancelled, this is for the correct module
-                                string output = result.StandardOutput;
-                                IReadOnlyList<GitItemStatus> changedFiles = _getAllChangedFilesOutputParser.Parse(output);
-
-                                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                                GitWorkingDirectoryStatusChanged?.Invoke(this, new GitWorkingDirectoryStatusEventArgs(changedFiles));
-                            }
-                        }
-                        catch (OperationCanceledException)
-                        {
-                            // No action
+                            await UpdateAsync();
                         }
                         catch
                         {
+                            // Try again once
                             try
                             {
-                                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                                // Avoid possible popups on every file changes
-                                CurrentStatus = GitStatusMonitorState.Stopped;
+                                await Task.Delay(InteractiveUpdateDelay);
+                                await UpdateAsync();
```

Comment:
Instead (in addition?) to retrying, reenable when switching repos in ScheduleNextInteractiveTime()?

I would probably preferred a failure counter instead of nested updates.

Reply:
I had to switch repos too often. In addition, it stopped working without any indication.

A failure counter is worth consideration.